test/integration/master/kube_apiserver_test.go

```diff
@@ -85,6 +86,48 @@ func TestRun(t *testing.T) {
 	}
 }
 
+func endpointReturnsStatusOK(client *kubernetes.Clientset, path string) bool {
+	res := client.CoreV1().RESTClient().Get().AbsPath(path).Do()
+	var status int
+	res.StatusCode(&status)
+	return status == http.StatusOK
+}
+
+func TestStartupSequenceHealthzAndReadyz(t *testing.T) {
+	hc := &delayedCheck{}
+	instanceOptions := &kubeapiservertesting.TestServerInstanceOptions{
+		InjectedHealthzChecker: hc,
+	}
+	server := kubeapiservertesting.StartTestServerOrDie(t, instanceOptions, []string{"--maximum-startup-sequence-duration", "5s"}, framework.SharedEtcd())
+	defer server.TearDownFn()
+
+	client, err := kubernetes.NewForConfig(server.ClientConfig)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	if endpointReturnsStatusOK(client, "/readyz") {
+		t.Fatalf("readyz should start unready")
+	}
+	// we need to wait longer than our grace period
+	err = wait.Poll(100*time.Millisecond, wait.ForeverTestTimeout, func() (bool, error) {
```

Comment:
but... but... you used a clock earlier?

Reply:
Yeah, I can rewrite this to leverage the clock.

Reply:
I am going to avoid doing this, because I can't think of a good way to plumb the clock through the  integration testserver.

Reply:
no clock magic here. This is an (e2e-like) integration test. We want the normal timing behaviour.

Reply:
👍